code/modules/medical/genetics/bioEffects/hemochromia.dm

```diff
@@ -0,0 +1,442 @@
+/datum/bioEffect/hemochromia
+	name = "Hemochromia Type-U"
+	desc = "A volatile mutation with 12 known stable alternatives. Will quickly break down into one of them if the subject exits the scanner."
+	id = "hemochromia_unknown"
+	probability = 111
+	effectType = EFFECT_TYPE_MUTANTRACE
+	msgGain = "You feel like you're out of breath."
+	msgLose = "You feel more stable."
+	reclaim_fail = 10
+	lockProb = 1
+	lockedGaps = 2
+	lockedDiff = 2
+	lockedChars = list("A","T","C","G")
+	lockedTries = 12
+	stability_loss = 5
+	icon_state  = "hemochromia_unknown"
+	var/typeRange = 0
+	var/duplicateCheck = 0
+	var/blood_color_R
+	var/blood_color_G
+	var/blood_color_B
+
+
+
+	OnLife()
+		if(ishuman(owner))
+			var/mob/living/carbon/human/H = owner
+			if(prob(12))
+				if(!istype(H.loc, /obj/machinery/genetics_scanner))
+					if(H.bioHolder.HasEffectInEither("hemochromia_rust" || "hemochromia_bronze" || "hemochromia_gold" || "hemochromia_lime" || "hemochromia_olive" || "hemochromia_jade" || "hemochromia_teal" || "hemochromia_cobalt" || "hemochromia_indigo" || "hemochromia_purple" || "hemochromia_violet" || "hemochromia_fuschia"))
+						duplicateCheck = 1
+					if(duplicateCheck != 0)
+						holder.RemoveEffect(src.id)
+						holder.RemovePoolEffect(src)
+					if(id == "hemochromia_unknown")
+						if(4 >= rand(13))
+							switch(H.bioHolder.bloodType)
+								if("A+")
+									if(pick(1, 2) == 1)
+										typeRange = 1
+									else
+										typeRange = 2
+								if("A-") typeRange = 3
+								if("B+") typeRange = 4
+								if("B-") typeRange = 5
+								if("AB+") typeRange = 6
+								if("O-") typeRange = 7
+								if("O+") typeRange = 8
+								if("AB-")
+									if(pick(1, 2) == 1)
+										typeRange = 9
+									else
+										typeRange = 10
+								else 
+									if(pick(1, 2, 3) == 1)
+										typeRange = 12
+									else
+										typeRange = 11
+						else
+							typeRange = rand(1,12)
+						var/datum/bioEffect/NEW = null
+						switch(typeRange)
```

Comment:
I personally think that the dependence of blood type makes little difference and this could all just be rewriten in approximately two lines if you decided to remove it. `NEW = new (pick(childrentypesof(/datum/bioEffect/hemochromia))()` or something like that.

Reply:
I like the dependence of blood type because it gives a light customization option. Like, RP players would probably appreciate it if they'd know that their character has some sort of "affinity" towards a certain color.